client/src/components/mapping/FhirMappingPanel/FhirResourceTree/index.tsx

```diff
@@ -219,6 +225,7 @@ const FhirResourceTree = ({ onClickCallback }: Props) => {
       node.childNodes = buildSlicedNode(node);
     } else if (attribute.choices.length > 0) {
       // if the node has choices, create a node for each of them
+      attribute.choices.forEach(choice => attribute.parent?.addChild(choice));
```

Comment:
I think this is the wrong place to do this. I get the issue: when we do Attribute.from avec having fetched the definition, we don't have access to the parent so `addChoice` doesn't properly set the parent. I think we should provide the parent to the `buildAttributes` function so that Attribute.from() does his job. we can talk about it, maybe i'm getting this wrong

Reply:
Yes, that was the problem
I tried to give an optional parent to `buildAttributes` but it didn't work properly not sure why. I can try again tomorrow
Also, this forEach is used just below in `handleContextMenu` so if we find a better way, we'll clean this at several places :)

Reply:
ok let's try again together tomorrow!